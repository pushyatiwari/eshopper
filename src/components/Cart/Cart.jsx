import React from 'react';
import './Cart.css';
import PropTypes from 'prop-types';
import CartItems from '../CartItems/CartItems';

const Cart = ({ description }) => (
  <CartItems description={description} />
);

const configShape = {
  id: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
  count: PropTypes.number,
  src: PropTypes.string,
};
Cart.propTypes = {
  description: PropTypes.arrayOf(PropTypes.shape(configShape)).isRequired,
};
export default Cart;
