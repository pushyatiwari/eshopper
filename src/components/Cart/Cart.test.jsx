import {
  render,
} from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Cart from './Cart';

describe(Cart.name, () => {
  const mockDescription = [{
    id: 'b2',
    name: 'melon',
    price: 40,
    count: 1,
    src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRRhOqYV2NWv0m80WzwGG2fCz6c3PzHmpz1Dg&usqp=CAU',

  }];
  test('should be called with', () => {
    const cartContainer = render(
      <BrowserRouter>
        <Cart
          description={mockDescription}
        />
      </BrowserRouter>,
    );
    expect(cartContainer).toMatchSnapshot();
  });
});
