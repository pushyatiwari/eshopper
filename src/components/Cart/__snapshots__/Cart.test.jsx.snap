// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should be called with 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cartitem-container"
      >
        <span
          class="basket-item"
        >
          YOUR BASKET (
          1 
          items)
        </span>
        <div
          class="basket-container"
        >
          <div
            class="upper-container"
          >
            <table>
              <tbody>
                <tr
                  class="cartitems-header"
                >
                  <td>
                    ITEM DESCRIPTION
                  </td>
                  <td>
                    UNIT PRICE
                  </td>
                  <td>
                    QUANTITY
                  </td>
                  <td>
                    SUBTOTAL
                  </td>
                </tr>
                <tr
                  class="cartitem-category"
                >
                  <td>
                    Fruits and Vegatables
                  </td>
                  <td />
                  <td />
                  <td />
                </tr>
                <tr>
                  <td>
                    melon
                  </td>
                  <td>
                    Rs.
                    40
                  </td>
                  <td>
                    1
                  </td>
                  <td>
                    40
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="bottom-part"
          >
            <a
              href="/"
            >
              <button
                class="continue-shopping"
                type="button"
              >
                CONTINUE SHOPPING
              </button>
            </a>
            <div
              class="cartitem-display-total"
            >
              <div>
                TOTAL
                 40
              </div>
              <hr />
              <div>
                <a
                  href="/checkout"
                >
                  <button
                    id="checkout"
                    type="button"
                  >
                    CHECKOUT
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cartitem-container"
    >
      <span
        class="basket-item"
      >
        YOUR BASKET (
        1 
        items)
      </span>
      <div
        class="basket-container"
      >
        <div
          class="upper-container"
        >
          <table>
            <tbody>
              <tr
                class="cartitems-header"
              >
                <td>
                  ITEM DESCRIPTION
                </td>
                <td>
                  UNIT PRICE
                </td>
                <td>
                  QUANTITY
                </td>
                <td>
                  SUBTOTAL
                </td>
              </tr>
              <tr
                class="cartitem-category"
              >
                <td>
                  Fruits and Vegatables
                </td>
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td>
                  melon
                </td>
                <td>
                  Rs.
                  40
                </td>
                <td>
                  1
                </td>
                <td>
                  40
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="bottom-part"
        >
          <a
            href="/"
          >
            <button
              class="continue-shopping"
              type="button"
            >
              CONTINUE SHOPPING
            </button>
          </a>
          <div
            class="cartitem-display-total"
          >
            <div>
              TOTAL
               40
            </div>
            <hr />
            <div>
              <a
                href="/checkout"
              >
                <button
                  id="checkout"
                  type="button"
                >
                  CHECKOUT
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
